src/AzureIoTHub.Portal/Server/Mappers/LoRaDeviceModelMapper.cs

```diff
@@ -54,14 +69,36 @@ public LoRaDeviceModel CreateDeviceModel(TableEntity entity)
+            entity[nameof(LoRaDeviceModel.IsOTAAsetting)] = model.IsOTAAsetting;
 
             var desiredProperties = new Dictionary<string, object>();
 
-            desiredProperties.Add($"properties.desired.{nameof(LoRaDeviceModel.AppEUI)}", model.AppEUI);
-            desiredProperties.Add($"properties.desired.{nameof(LoRaDeviceModel.SensorDecoder)}", model.SensorDecoder);
+            if (model.IsOTAAsetting)
+            {
+                desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceModel.AppEUI), model.AppEUI, desiredProperties);
+            }
+
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceModel.SensorDecoder), model.SensorDecoder, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.Supports32BitFCnt), model.Supports32BitFCnt, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.ABPRelaxMode), model.ABPRelaxMode, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.KeepAliveTimeout), model.KeepAliveTimeout, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.PreferredWindow), model.PreferredWindow, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.Downlink), model.Downlink, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.Deduplication), model.Deduplication, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.FCntDownStart), model.FCntDownStart, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.FCntResetCounter), model.FCntResetCounter, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.FCntUpStart), model.FCntUpStart, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.RX1DROffset), model.RX1DROffset, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.RX2DataRate), model.RX2DataRate, desiredProperties);
+            desiredProperties = this.AddOptionnalProperties(entity, nameof(LoRaDeviceBase.RXDelay), model.RXDelay, desiredProperties);
+
+            return desiredProperties;
+        }
+
+        private Dictionary<string, object> AddOptionnalProperties(TableEntity entity, string propertyName, object propertyValue, Dictionary<string, object> desiredProperties)
+        {
+            entity[propertyName] = propertyValue;
+            desiredProperties.Add($"properties.desired.{propertyName}", propertyValue);
 
             return desiredProperties;
         }
```

Comment:
No need to return the dictionary since it's a value reference in the parameter.
```suggestion
        private void AddOptionnalProperties(TableEntity entity, string propertyName, object propertyValue, Dictionary<string, object> desiredProperties)
        {
            entity[propertyName] = propertyValue;
            desiredProperties.Add($"properties.desired.{propertyName}", propertyValue);
        }
```